src/api/queues.js

```diff
@@ -40,29 +40,23 @@ router.get(
       'defaultScope',
       'questionCount'
     ).findAll()
-    if (userAuthz.isAdmin) {
-      const queues = queuesResult.map(queue => queue.get({ plain: true }))
-      res.json(queues)
-    } else {
-      const courses = await Course.findAll()
-      const filteredCourseIds = courses
-        .filter(course => {
-          const courseId = course.id
-          if (
-            !course.isUnlisted ||
-            userAuthz.staffedCourseIds.indexOf(courseId) !== -1
-          ) {
-            return true
-          }
-          return false
-        })
-        .map(course => course.id)
-      const filteredQueues = queuesResult.filter(queue =>
-        filteredCourseIds.includes(queue.courseId)
+
+    const filteredCourseIds = await Course.findAll()
```

Comment:
If it wasn't possible to do this and other queries with Sequelize itself (i.e. filtering in the DB and not after reading everything) could you document why that was the case for future maintainers?

Reply:
So I think I came up with a way to do it all in a sequelize query (I was overcomplicating things earlier): 
`if (userAuthz.isAdmin) { 
      // sequelize query to find all courses 
} else { 
     // sequelize query to find all unlisted courses
     // sequelize query to find all courses that user is a "staff" for that are not unlisted 
     // join the above two sequelize queries and those are all of the courses shown
}`

I'm gonna try to implement this, and hopefully this works; if not, I'll document why that didn't work

Reply:
Sounds good! This may be asking too much of sequelize, but my hope was for a) a query for all courses for admins, and b) a separate query that uses some set of `SELECT`, `JOIN`, `WHERE`, etc. to do all the filtering and joining entirely in the DB. I'm not entirely sure how to express these constraints with the ORM though.

Reply:
Surprisingly, I think I managed to get a query for the filtering to work - it's basically a `WHERE` with an `OR` that checks whether id is in staffedCourseIds or if the course is not unlisted.

Reply:
Love it!